packages/main/src/plugin/container-registry.ts

```diff
@@ -588,6 +588,49 @@ export class ContainerProviderRegistry {
     return flattenedImages;
   }
 
+  async podmanListImages(options?: PodmanListImagesOptions): Promise<ImageInfo[]> {
+    let telemetryOptions = {};
+
+    // This gets all the available providers if no provider has been specified
+    let providers: InternalContainerProvider[];
+    if (options?.provider === undefined) {
+      providers = Array.from(this.internalProviders.values());
+    } else {
+      providers = [this.getMatchingContainerProvider(options?.provider)];
+    }
+
+    const images = await Promise.all(
+      Array.from(providers).map(async provider => {
+        try {
+          // This is important and very similar to the 'listImages' function with the difference
+          // is that we are using the libpod API to list images, so we only retrieve the images
+          // from providers that have implemented libpod API.
+          if (!provider.libpodApi) {
+            console.log('podman engine is missing libpod API, cannot list images, check your provider configuration');
+            return [];
+          }
+
+          // List the images
+          const images = await provider.libpodApi.podmanListImages(options);
+
+          // Add the additional information such as engineName and engineId to the ImageInfo
+          return images.map(image => {
+            const imageInfo: ImageInfo = { ...image, engineName: provider.name, engineId: provider.id };
+            return imageInfo;
+          });
+        } catch (error) {
+          console.log('error in engine', provider.name, error);
```

Comment:
nit: should be log.error call. Message should be more specific and mention that provider failed to list images.